power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/iterative_linear_se_solver.hpp

```diff
@@ -768,10 +842,10 @@ template <bool sym> class IterativeLinearSESolver {
                 // shunt
                 if (type == YBusElementType::shunt) {
                     if (measured_value.has_shunt(obj)) {
-                        SensorCalcParam<sym> const& m = measured_value.shunt_power(obj);
+                        PowerSensorCalcParam<sym> const& m = measured_value.shunt_power(obj);
                         // eta -= Ys^H * i_shunt / variance
```

Comment:
This should be

`(variance)^-1 * Ys^H * i_shunt`

Reply:
Oh, for this one it is actually the same. `Ys^H * i_shunt` and `variance` are both vectors, so the order of multiplication does not matter.

Reply:
`Ys^H * (variance)^-1 * i_shunt`, right?

Reply:
eigen::DiagonalWrapper is a rabbithole that i'll need to dive further into. I've tried going down this route before (also relating to how we implement `dot`) but I will try it again